code/go/0chain.net/smartcontract/minersc/miner.go

```diff
@@ -193,28 +193,20 @@ func (msc *MinerSmartContract) deleteNode(
 ) (*MinerNode, error) {
 	var err error
 	deleteNode.Delete = true
-	var nodeType spenum.Provider
 	switch deleteNode.NodeType {
 	case NodeTypeMiner:
-		nodeType = spenum.Miner
 	case NodeTypeSharder:
-		nodeType = spenum.Sharder
 	default:
 		return nil, fmt.Errorf("unrecognised node type: %v", deleteNode.NodeType.String())
 	}
 
-	for key, pool := range deleteNode.Pools {
+	for _, pool := range deleteNode.Pools {
 		switch pool.Status {
-		case spenum.Pending:
-			_, err := deleteNode.UnlockPool(
-				pool.DelegateID, nodeType, key, balances)
-			if err != nil {
-				return nil, fmt.Errorf("error emptying delegate pool: %v", err)
-			}
 		case spenum.Active:
 			pool.Status = spenum.Deleting
-		case spenum.Deleting:
-		case spenum.Deleted:
+		case spenum.Pending,
+			spenum.Deleting,
```

Comment:
why can't we completely remove Deleting status? I don't see a lot of logic built on this status, most likely ``Unstaking`` is what we need for 2 phase unlocking and use. Deleting is used in ``payFees`` and ``MintRewards`` but is doesn't look as something important.

Reply:
Yes, it should be safe to be removed completely